src/core/server/elasticsearch/client/configure_client.ts

```diff
@@ -137,14 +139,46 @@ const addLogging = (client: Client, logger: Logger) => {
             http: { request: { id: event.meta.request.options.opaqueId } },
           }
         : undefined; // do not clutter logs if opaqueId is not present
+      let queryMessage = '';
       if (error) {
         if (error instanceof errors.ResponseError) {
-          logger.debug(`${getResponseMessage(event)} ${getErrorMessage(error)}`, meta);
+          queryMessage = `${getResponseMessage(event)} ${getErrorMessage(error)}`;
         } else {
-          logger.debug(getErrorMessage(error), meta);
+          queryMessage = getErrorMessage(error);
         }
       } else {
-        logger.debug(getResponseMessage(event), meta);
+        queryMessage = getResponseMessage(event);
+      }
+
+      queryLogger.debug(queryMessage, meta);
+
+      if (event.warnings && event.warnings.length > 0) {
+        // Plugins can explicitly mark requests as originating from a user by
+        // removing the `'x-elastic-product-origin': 'kibana'` header that's
+        // added by default. User requests will be shown to users in the
+        // upgrade assistant UI as an action item that has to be addressed
+        // before they upgrade.
+        // Kibana requests will be hidden from the upgrade assistant UI and are
+        // only logged to help developers maintain their plugins
+        const requestOrigin =
+          (event.meta.request.options.headers != null &&
+            (event.meta.request.options.headers[
+              'x-elastic-product-origin'
+            ] as unknown as string)) === 'kibana'
+            ? 'kibana'
+            : 'user';
+
+        const stackTrace = new Error().stack?.split('\n').slice(5).join('\n');
+
+        if (requestOrigin === 'kibana') {
+          deprecationLogger.warn(
+            `ES DEPRECATION: ${event.warnings}\nOrigin:${requestOrigin}\nStack trace:\n${stackTrace}\nQuery:\n${queryMessage}`
+          );
+        } else {
+          deprecationLogger.error(
+            `ES DEPRECATION: ${event.warnings}\nOrigin:${requestOrigin}\nStack trace:\n${stackTrace}\nQuery:\n${queryMessage}`
+          );
```

Comment:
NIT: as you did with `queryMessage` upper in the file

```ts
const deprecationMsg = `ES DEPRECATION: ${event.warnings}\nOrigin:${requestOrigin}\nStack trace:\n${stackTrace}\nQuery:\n${queryMessage}`;
if (requestOrigin === 'kibana') {
  deprecationLogger.warning(deprecationMsg);
} else {
  deprecationLogger.error(deprecationMsg);
}

```

Reply:
Luke raised the same thing, but we have `${requestOrigin}` in the string, so it's not just the level of logging.